### PR TITLE
test: get all backend tests into a working state; run tests in CI

### PR DESCRIPTION
**Pull request by hallettj**

### CODE REVIEW

File: template/packages/backend/jest-unit.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (hallettj) [inline template/packages/backend/jest-unit.json:1]: I moved this config out of `package.json`

File: template/packages/backend/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  projects: ["jest-unit.json", "test/jest-e2e.json"],
+}
```
Comment (hallettj) [inline template/packages/backend/jest.config.js:3]: Jest comes through! Using the `projects` feature jest can run multiple test configs with one runner which makes it easy to run e2e and unit tests with one command.

File: template/packages/backend/src/app.module.ts [hunk 3]
```diff
@@ -45,7 +43,6 @@ const config = new ConfigService()
       synchronize: config.isDevDatabase,
     }),
   ],
-  controllers: [AppController],
-  providers: [AppService, { provide: ConfigService, useValue: config }],
+  providers: [{ provide: ConfigService, useValue: config }],
```
Comment (hallettj) [inline template/packages/backend/src/app.module.ts:46]: I pulled out the example REST controller

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            ${{ runner.os }}-node-
+
+      - name: Cache TypeScript output
+        uses: actions/cache@v1
+        with:
+          path: |
+            template/packages/backend/dist
+          key: ignored
+
+      - run: yarn --frozen-lockfile
```
Comment (xasopheno) [inline .github/workflows/test.yml:29]: --frozen-lockfile is great. very cool.

File: template/packages/backend/package.json [hunk 5]
```diff
@@ -23,9 +23,11 @@
     "lint": "eslint \"{src,apps,libs,test}/**/*.ts\"",
     "test": "jest",
     "test:watch": "jest --watch",
-    "test:cov": "jest --coverage",
+    "test:cov": "jest --selectProjects unit --coverage",
```
Comment (xasopheno) [inline template/packages/backend/package.json:26]: neat

File: template/packages/frontend/package.json [hunk 6]
```diff
@@ -9,10 +9,11 @@
     "start:dev": "concurrently -k 'next dev' 'yarn codegen:watch'",
     "codegen": "ts-graphql-plugin typegen",
     "codegen:watch": "nodemon --ext ts,tsx --exec yarn codegen --ignore '/__generated__/'",
-    "typecheck": "echo next && tsc -p .",
+    "typecheck": "echo frontend && tsc --pretty",
     "typecheck:watch": "yarn typecheck --watch",
     "lint": "eslint --cache --ext .ts,.tsx --ignore-pattern __generated__ .",
-    "storybook": "start-storybook -p 6006 -c .storybook"
+    "storybook": "start-storybook -p 6006 -c .storybook",
+    "test": "echo 'TODO: frontend tests'"
```
Comment (xasopheno) [inline template/packages/frontend/package.json:16]: :) Noted.

File: template/packages/backend/test/app.e2e-spec.ts [hunk 7]
```diff
@@ -1,24 +1,96 @@
...
+import {
+  Config,
+  startPostgresContainer,
+} from "@originate/docker-await-postgres"
+import request from "supertest"
+import { createConnection } from "typeorm"
 
-describe('AppController (e2e)', () => {
-  let app: INestApplication;
+jest.setTimeout(15_000)
```
Comment (xasopheno) [inline template/packages/backend/test/app.e2e-spec.ts:10]: Do you think we still need this timeout? I remember we added when were trying to solve a problem. Are you expecting that tests on CI might take a while if they are spinning up a docker container?

File: template/packages/backend/src/app.module.ts [hunk 8]
```diff
@@ -45,7 +43,6 @@ const config = new ConfigService()
       synchronize: config.isDevDatabase,
     }),
   ],
-  controllers: [AppController],
-  providers: [AppService, { provide: ConfigService, useValue: config }],
```
Comment (xasopheno) [inline template/packages/backend/src/app.module.ts:49]: Why do we no longer need the AppService and the AppController? Was this just a boilerplate module we are removing?

File: template/packages/backend/src/recipes/recipe.service.spec.ts [hunk 9]
```diff
@@ -1,18 +1,49 @@
-import { Test, TestingModule } from '@nestjs/testing';
-import { RecipeService } from './recipe.service';
+import { Test, TestingModule } from "@nestjs/testing"
+import { getEntityManagerToken, getRepositoryToken } from "@nestjs/typeorm"
```
Comment (xasopheno) [inline template/packages/backend/src/recipes/recipe.service.spec.ts:2]: Just learning about these concept of database tokens in nest. Neat stuff.

**Review by xasopheno — APPROVED** [review]
LGTM. It's so nice to have examples of how all these kinds of tests should look. So helpful! It's hard to not write a good test when there's already an example of how to do it.

File: template/packages/backend/test/app.e2e-spec.ts (hunk 7 above)
Comment (hallettj) [reply template/packages/backend/test/app.e2e-spec.ts:10]: Yes, I think we still need this. The default timeout is 5 seconds, and that's not enough on my machine. We still haven't tested on OS X.

File: template/packages/backend/src/app.module.ts (hunk 8 above)
Comment (hallettj) [reply template/packages/backend/src/app.module.ts:49]: AppController was an example of how to write a REST service. My thinking is if we're pushing GraphQL we don't need a REST example.

**Pull request merged**